bmm/src/main/java/org/openehr/bmm/v2/persistence/PBmmClass.java

```diff
@@ -125,12 +164,76 @@ public void setSourceSchemaId(String sourceSchemaId) {
         this.sourceSchemaId = sourceSchemaId;
     }
 
-    public Map<String, PBmmType> getAncestorDefs() {
+    public Map<String, PBmmUnitaryType> getAncestorDefs() {
         return ancestorDefs;
     }
 
-    public void setAncestorDefs(Map<String, PBmmType> ancestorDefs) {
-        this.ancestorDefs = ancestorDefs;
+    public BmmClass createBmmClass() {
+        BmmClass bmmClass;
+        if (getGenericParameterDefs().size() > 0) {
+            bmmClass = new BmmGenericClass(getName(), getDocumentation(), nullToFalse(isAbstract()));
+        } else {
+            bmmClass = new BmmSimpleClass(getName(), getDocumentation(), nullToFalse(isAbstract()));
+        }
+        bmmClass.setOverride(nullToFalse(isOverride()));
+        bmmClass.setSourceSchemaId(getSourceSchemaId());
+        return bmmClass;
+    }
+
+    public BmmClass populateBmmClass(BmmClassProcessor bmmModel, PBmmSchema schema) {
```

Comment:
Maybe rename the `BmmClassProcessor` parameter to `classProcessor`, similar to `populateBmmClassProperties`?

```suggestion
    public BmmClass populateBmmClass(BmmClassProcessor classProcessor, PBmmSchema schema) {
```